ondine/jse-win15-1.md

```diff
@@ -0,0 +1,216 @@
+
+**k)** `typeof (x % 2) === "number"`
+
+Always true. Whatever x is, the type of the remainder is a number.
+
+**l)** `x++ == ++x`
+
+Always false. x++ evaluates to unincremented value of x. ++x evaluates to the incremented value of x.
+
+**m)** `++x == x++`
+
+False if x is not a number value. True if x is a number, because the incremented value of x on the left equals the unincremented value on the right.
+
+**n)** `"1"+x == 1+x`
+
+Only true if x is a string. With any other value, type coercion returns a concatenated string "1value".
+
+**o)** `"0"+1 == 1`
+
+True. Type conversion. "0" converts to 0, then addition makes the result true.
+
+Or - "0" as a string is truthy, numbers other than 0 are truthy.
+
+**p)** `(typeof (x+1))===(typeof x)`
+
+True if the value of x is a number, a string or NaN because the type is "number" for both sides.
+
+False if x is a Boolean, undefined, or null because (x + 1) converts its type to "number" while x is a type of "boolean", "undefined", or "object", respectively.
+
+**q)** `(x*1 == x) || ((typeof x) != "number")`
+
+True. If x is a number, then the left side is true. If anything else other than NaN, the typeof x is not a number.
+
+False only if x is NaN, because then the type of x is a number, and the left side is false.
+
+**r)** `(x=(typeof (x+(typeof x))))==x`
+
+Always true. Whatever the left side evaluates to, x is assigned that value, which is equal to x.
+
+---
+
+All of the following can be solved with ordinary expressions and global variables with primitive values. You don't need functions, loops, or other topics beyond our first two classes.
+
+ **2)** (_Difficulty: easy_)
+
+Assume variables x, y, and z are numbers.
+
+**a)**
+Write an expression for the mean (i.e. average) of x, y, and z.
+
+`(x + y + z)/3`
+
+**b)**
+Write a series of expressions to adjust each of x, y, and z halfway toward their mean.
+That is, reset the value of each variable to something based on its previous value.
+
+```
+var mean = (x + y + z)/3;
+x = (mean-x)/2 + x;
+y = (mean-y)/2 + y;
+z = (mean-z)/2 + z;
+```
+---
+
+**3)** (_Difficulty: easyish_)
+
+Suppose you're encoding geometric shapes in a Cartesian (2D) coordinate system, and you represent a rectangle with four numeric variables:
+
+- _l_ : horizontal position of left edge;
+- _r_ : horizontal position of right edge;
+- _t_ : vertical position of top edge;
+- _b_ : vertical position of bottom edge.
+
+**a)**
+Write an expression for the rectangle's area.
+
+`var areaRect = (r - l) * (t - b)`
+
+**b)**
+Write an expression which is true if the rectangle is taller than it is wide, and false otherwise.
+
+`(t - b) > (r - l)`
+
+**c)**
+Write an expression for the circumference of the biggest circle which can fit inside the rectangle.  (Hint: you'll need logic similar to that in **b**.)
+
+```
+if ((r - l) > (t - b)) { 
+  var diameter = (t - b); 
+  } else { 
+  diameter = (r - l)
+ 
+}
```

Comment:
Good! The only thing I'd point out here is that you're only defining `diameter` in the first branch of your `if` statement. If that isn't the one that's run, the `diameter = (r - l)` will set a global variable called `diameter`, which is not what you want. The easiest way around this is to declare `var diameter;` before your `if` statement, and then assign a value to that variable inside the `if` branches.